internal/wasm/jit/asm/impl.go

```diff
@@ -0,0 +1,47 @@
+package asm
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+)
+
+type BaseAssemblerImpl struct {
+	// setBranchTargetOnNextInstructions holds branch kind instructions (BR, conditional BR, etc)
+	// where we want to set the next coming instruction as the destination of these BR instructions.
+	setBranchTargetOnNextNodes []Node
+	// onGenerateCallbacks holds the callbacks which are called after generating native code.
+	OnGenerateCallbacks []func(code []byte) error
+}
+
+// SetJumpTargetOnNext implements AssemblerBase.SetJumpTargetOnNext
+func (a *BaseAssemblerImpl) SetJumpTargetOnNext(nodes ...Node) {
+	a.setBranchTargetOnNextNodes = append(a.setBranchTargetOnNextNodes, nodes...)
+}
+
+// AddOnGenerateCallBack implements AssemblerBase.AddOnGenerateCallBack
+func (a *BaseAssemblerImpl) AddOnGenerateCallBack(cb func([]byte) error) {
+	a.OnGenerateCallbacks = append(a.OnGenerateCallbacks, cb)
+}
+
+// BuildJumpTable implements AssemblerBase.BuildJumpTable
+func (a *BaseAssemblerImpl) BuildJumpTable(table []byte, labelInitialInstructions []Node) {
+	a.AddOnGenerateCallBack(func(code []byte) error {
+		// Build the offset table for each target.
+		base := labelInitialInstructions[0].OffsetInBinary()
+		for i, nop := range labelInitialInstructions {
+			if uint64(nop.OffsetInBinary())-uint64(base) >= math.MaxUint32 {
```

Comment:
can we define a constant in like internal/limits.go and move this comment there?